Later binutils require need code sections to be aligned. Also changed from daft assembler
syntax  to something more sensible
--- libc/nptl/sysdeps/unix/sysv/linux/st200/pt-initfini.c.orig	2007-05-30 12:49:02.000000000 +0100
+++ libc/nptl/sysdeps/unix/sysv/linux/st200/pt-initfini.c	2007-05-30 12:49:19.000000000 +0100
@@ -57,7 +57,8 @@
 "/*@_init_PROLOG_ENDS*/                                                     \n"
 "                                                                           \n"
 "/*@_init_EPILOG_BEGINS*/                                                   \n"
-"	.section .init                                                      \n"
+"       .section .init, \"ax\", @progbits                                   \n"
+"	.align 4                                                            \n"
 "       ldw     $r63 = 16[sp]                                               \n"
 "       ;;                                                                  \n"
 "       ldw     gp = 20[sp]                                                 \n"
@@ -86,7 +87,8 @@
 "	;;                                                                  \n"
 "                                                                           \n"
 "/*@_fini_EPILOG_BEGINS*/                                                   \n"
-"	.section .fini                                                      \n"
+"	.section .fini, \"ax\", @progbits                                   \n"
+"	.align 4                                                            \n"
 "       ldw     $r63 = 16[sp]                                               \n"
 "       ;;                                                                  \n"
 "       ldw     gp     = 20[sp]                                             \n"
--- libc/sysdeps/st200/elf/initfini.c.orig	2007-05-30 12:49:10.000000000 +0100
+++ libc/sysdeps/st200/elf/initfini.c	2007-05-30 12:49:19.000000000 +0100
@@ -30,8 +30,8 @@
 __asm__ ("\n\n"
 "#include \"defs.h\"                                                        \n"
 "/*define SHARED*/                                                          \n"
-"#define sp $r0.12                                                          \n"
-"#define gp $r0.14                                                          \n"
+"#define sp $r12                                                            \n"
+"#define gp $r14                                                            \n"
 "                                                                           \n"
 "/*@HEADER_ENDS*/                                                           \n"
 "                                                                           \n"
@@ -46,36 +46,37 @@
 "_init:                                                                     \n"
 "       add     sp    = sp, -32                                             \n"
 "       ;;                                                                  \n"
-"       stw     16[sp] = $r0.63                                             \n"
+"       stw     16[sp] = $r63                                               \n"
 "       ;;                                                                  \n"
 "       stw     20[sp] = gp                                                 \n"
 "       ;;                                                                  \n"
-"       call    $r0.63 = 0f                                                 \n"
+"       call    $r63 = 0f                                                   \n"
 "       ;;                                                                  \n"
 "0:                                                                         \n"
-"       add     gp = $r0.63, @neggprel(0b)                                  \n"
+"       add     gp = $r63, @neggprel(0b)                                    \n"
 "       ;;                                                                  \n"
 "	.weak	__gmon_start__                                              \n"
-"       ldw     $r0.63 = @gotoff(__gmon_start__)[gp]                        \n"
+"       ldw     $r63 = @gotoff(__gmon_start__)[gp]                          \n"
 "       ;;                                                                  \n"
-"       cmpeq   $b0.0  = $r0.63, 0                                          \n"
+"       cmpeq   $b0  = $r63, 0                                              \n"
 "       ;;                                                                  \n"
-"       br      $b0.0, 1f                                                   \n"
+"       br      $b0, 1f                                                     \n"
 "       ;;                                                                  \n"
-"       call    $r0.63 = $r0.63                                             \n"
+"       call    $r63 = $r63                                                 \n"
 "       ;;                                                                  \n"
 "1:                                                                         \n"
 "/*@_init_PROLOG_ENDS*/                                                     \n"
 "                                                                           \n"
 "/*@_init_EPILOG_BEGINS*/                                                   \n"
-"	.section .init                                                      \n"
-"       ldw     $r0.63 = 16[sp]                                             \n"
+"	.section .init, \"ax\", @progbits                                   \n"
+"	.align 4                                                            \n"
+"       ldw     $r63 = 16[sp]                                               \n"
 "       ;;                                                                  \n"
 "       ldw     gp     = 20[sp]                                             \n"
 "       ;;                                                                  \n"
 "       add     sp     = sp, 32                                             \n"
 "       ;;                                                                  \n"
-"       goto    $r0.63                                                      \n"
+"       goto    $r63                                                        \n"
 "       ;;                                                                  \n"
 "/*@_init_EPILOG_ENDS*/                                                     \n"
 "                                                                           \n"
@@ -86,29 +87,30 @@
 "_fini:                                                                     \n"
 "       add     sp    = sp, -32                                             \n"
 "       ;;                                                                  \n"
-"       stw     16[sp] = $r0.63                                             \n"
+"       stw     16[sp] = $r63                                               \n"
 "       ;;                                                                  \n"
 "       stw     20[sp] = gp                                                 \n"
 "       ;;                                                                  \n"
-"       call    $r0.63 = 0f                                                 \n"
+"       call    $r63 = 0f                                                   \n"
 "       ;;                                                                  \n"
 "0:                                                                         \n"
-"       add     gp = $r0.63, @neggprel(0b)                                  \n"
+"       add     gp = $r63, @neggprel(0b)                                    \n"
 "       ;;                                                                  \n"
 "                                                                           \n"
 "/*@_fini_PROLOG_ENDS*/                                                     \n"
-"	call    $r0.63 = i_am_not_a_leaf                                    \n"
+"	call    $r63 = i_am_not_a_leaf                                      \n"
 "	;;                                                                  \n"
 "                                                                           \n"
 "/*@_fini_EPILOG_BEGINS*/                                                   \n"
-"	.section .fini                                                      \n"
-"       ldw     $r0.63 = 16[sp]                                             \n"
+"	.section .fini, \"ax\", @progbits                                   \n"
+"	.align 4                                                            \n"
+"       ldw     $r63 = 16[sp]                                               \n"
 "       ;;                                                                  \n"
 "       ldw     gp     = 20[sp]                                             \n"
 "       ;;                                                                  \n"
 "       add     sp     = sp, 32                                             \n"
 "       ;;                                                                  \n"
-"       goto    $r0.63                                                      \n"
+"       goto    $r63                                                        \n"
 "       ;;                                                                  \n"
 "                                                                           \n"
 "/*@_fini_EPILOG_ENDS*/                                                     \n"
